api/src/main/java/org/pmiops/workbench/cohortbuilder/QueryBuilderFactory.java

```diff
@@ -0,0 +1,45 @@
+package org.pmiops.workbench.cohortbuilder;
+
+import org.pmiops.workbench.cohortbuilder.querybuilder.AbstractQueryBuilder;
+import org.pmiops.workbench.exceptions.BadRequestException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Factory class that autowires different implementations of {@link AbstractQueryBuilder}
+ * and stores them on a map. {@link org.pmiops.workbench.cohortbuilder.querybuilder.FactoryKey}
+ * should be used to get an instance of {@link AbstractQueryBuilder}.
+ */
+@Component
+public class QueryBuilderFactory {
+
+    @Autowired
```

Comment:
Where does this binding come from?

Reply:
Spring will autowire any class that extends AbstractQueryBuilder into this list. Then @PostConstruct will be called after QueryBuilderFactory is created to populate the queryBuilderCache